### ISSUE TITLE
Data Import Advanced Mapping Requires Permission for Allocation's Payment Field

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/NPSP#5019]
Customers who've switched to Advanced Mapping for Batch Data Import will need to grant permissions to the GAU Allocation's npsp__Payment__c field, even though that field is related to non-GA functionality. This is required to process a Batch Gift Entry batch at the moment. Help Text Mapping does not require access to this field.

![bdi fls](https://user-images.githubusercontent.com/29731456/73361772-9ebbc700-4273-11ea-99f7-714ff23a08d4.png)

When you click the "Process Batch" button, an error appears:
![Screen Shot 2020-01-29 at 8 54 25 AM](https://user-images.githubusercontent.com/29731456/73362633-31109a80-4275-11ea-9cae-580944938a2a.png)

### Steps to Repeat:
1. Go to Setup --> Profiles --> System Administrator and remove permissions for the GAU Allocation's Payment field.
2. Go to the Batch Gift Entry tab and enter a gift in the batch. No errors are displayed in the list of gifts to be processed.
3. Go to NPSP Settings and enable Advanced Mapping.
4. Go to the Batch Gift Entry tab, create a new batch, and enter a gift in the batch. You'll see an error displayed next to each gift in the gift list that says "You do not have permissions to access npsp__Allocation.npsp__Payment__c."
5. Click the Process Batch button. A red banner appears at the top saying "There are errors on one or more entries. You must resolve all errors before you can process this batch."

### Workaround

- Grant permissions to the GAU Allocation's Payment field.

**Comment by jesslopez-sf** [comment]
**lurch: attach W-038446

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HqKZQA0" target="_blank">W-038446</a>

**Comment by jesslopez-sf** [comment]
Will be fixed in forthcoming feature

**Comment by mpietruszka** [comment]
Hello! I'm seeing this was closed as pending forthcoming fix, but we just encountered this in our org. (This is the correct way to notify this still seems to be outstanding, right?)

Additionally - the error message & steps described above don't clarify what level of access is necessary to get past the error - Read, or Read & Edit. I granted both in our org to our relevant import profiles so that we could be unblocked, but if only Read is necessary, I'd like to reduce that back. And if this gets fixed and the perms are no longer needed at all, I'd like to remove them entirely. Thanks!

**Comment by mbeaty-sf** [comment]
Hi @mpietruszka,

We don't provide support via GitHub Issues. Please post on the [Trailblazer Community](https://trailhead.salesforce.com/trailblazer-community/). Alternatively, please [Log a Case with Salesforce Support](https://powerofus.force.com/s/article/SFDO-CS-Create-a-Tech-Support-Case).
